findbugs/src/java/edu/umd/cs/findbugs/detect/FindRefComparison.java

```diff
@@ -1072,6 +1090,34 @@ private void handleSuspiciousRefComparison(JavaClass jclass, Method method, Meth
                 sourceLineAnnotation, location));
     }
 
+    private void handleValueBasedRefComparison(JavaClass jclass, Method method, MethodGen methodGen,
+            List<WarningWithProperties> refComparisonList, Location location, String signature, ReferenceType lhsType,
+            ReferenceType rhsType) {
+        XField xf = null;
+        if (lhsType instanceof FinalConstant) {
+            xf = ((FinalConstant) lhsType).getXField();
+        } else if (rhsType instanceof FinalConstant) {
+            xf = ((FinalConstant) rhsType).getXField();
+        }
+        String sourceFile = jclass.getSourceFileName();
+        String bugPattern = "VBC_REF_COMPARISON";
```

Comment:
The FindRefComparison detector must be marked in findbugs.xml as reporting "VBC_REF_COMPARISON" bug pattern